admin/data/sql/migrations/20191210082105.do.mod-vewpupillivecheckstakencount-new-pupil-state.sql

```diff
@@ -0,0 +1,31 @@
+ALTER VIEW [mtc_admin].[vewPupilLiveChecksTakenCount] AS
+
+    SELECT
+        p.id as pupil_id,
+        p.foreName,
+        p.middleNames,
+        p.lastName,
+        p.dateOfBirth,
+        p.urlSlug,
+        p.school_id,
+        count(*) as totalCheckCount
+    FROM
+        [mtc_admin].[pupil] p
+            INNER JOIN [mtc_admin].[check] AS chk ON (p.id = chk.pupil_id)
+            INNER JOIN [mtc_admin].[checkStatus] AS chkStatus ON (chk.checkStatus_id = chkStatus.id)
+    WHERE
+             -- ignore pupils who are not attending
+             p.attendanceId IS NULL
+             -- pupils must have already attempted 1 or more checks that are not expired
+      AND    chk.pupilLoginDate IS NOT NULL
+      AND    chkStatus.code <> 'EXP'
```

Comment:
@GuyHarwood @GeorgeChatzigiannis - do we want to remove the EXP status?  We can use the checkPin.expiresAt to determine non-expiry or the checkPin record will removed in 2020, to free up the pin for re-use, which will indicate either expiry or incomplete (so with pupilLoginDate we can decide).

Reply:
yes, much prefer to remove EXP and determine via the date stamp

Reply:
agreed with @GuyHarwood

Reply:
I have raised a new ticket to remove it.  Let's discuss privately.